Replace ioutil with os and io for the test

ahrtr opened the pull request:
The ioutil package has already been deprecated in golang 1.16, please see [go1.16#ioutil](https://golang.org/doc/go1.16#ioutil). So we need to replace all the ioutil with os and io.

This PR is just to update the test. There will be a couple of separate PRs for another SIGs. 

Previously there is a [PR](https://github.com/kubernetes/kubernetes/pull/105994), which is huge. So I am splitting the huge PR into a couple of smaller PRs. This PR is just one of them. 

Fix https://github.com/kubernetes/kubernetes/issues/100367

ahrtr committed:
io/ioutil has already been deprecated in golang 1.16, so replace all ioutil with io and os

fedebongio commented:
/remove-sig api-machinery

spiffxp commented:
/ok-to-test

SergeyKanzhelev commented:
Tests are not particularly affected by this comment: https://github.com/kubernetes/kubernetes/issues/100367#issuecomment-802591356 so it is probably all right to switch.

/priority backlog
/triage accepted

The PR is huge so already has merge conflicts. Some synchronization will be needed between getting approvals first, then rebasing and lgtm right away.

/approve

for node

ahrtr commented:
> Tests are not particularly affected by this comment: [#100367 (comment)](https://github.com/kubernetes/kubernetes/issues/100367#issuecomment-802591356) so it is probably all right to switch.
> 
> /priority backlog /triage accepted
> 
> The PR is huge so already has merge conflicts. Some synchronization will be needed between getting approvals first, then rebasing and lgtm right away.
> 
> /approve
> 
> for node

Thanks for the comment. Rebasing done!

spiffxp commented:
/kind cleanup
/release-note-none

spiffxp commented on test/images/apparmor-loader/loader.go:
```diff
@@ -118,7 +117,7 @@ func loadNewProfiles() (success bool, newProfiles []string) {
 	success = true
 	for _, dir := range dirs {
-		infos, err := ioutil.ReadDir(dir)
+		infos, err := os.ReadDir(dir)
```
same thing, you need to update `test/images/apparmor-loader/VERSION`

spiffxp commented on test/images/metadata-concealment/check_metadata_concealment.go:
```diff
@@ -128,7 +128,7 @@ func checkURL(url string, header http.Header, expectedStatus int, expectedToCont
 		return fmt.Errorf("unexpected response: got %d, want %d", resp.StatusCode, expectedStatus)
 	}
-	body, err := ioutil.ReadAll(resp.Body)
+	body, err := io.ReadAll(resp.Body)
```
same thing, you need to update `test/images/metadata-concealment/VERSION`

spiffxp commented on test/images/pets/peer-finder/peer-finder.go:
```diff
@@ -77,7 +77,7 @@ func shellOut(sendStdin, script string) {
 	stdin.Close()
 
-	out, err := ioutil.ReadAll(stdout)
+	out, err := io.ReadAll(stdout)
```
same thing, you need to update `test/images/peer-finder/VERSION`

spiffxp reviewed (changes requested):
I would suggest removing the test/images/** changes from this PR and doing those in another PR with the VERSION bumps I've asked for. The rest of this LGTM, though I agree its size makes this PR prone to getting stuck in rebases

ahrtr commented:
> I would suggest removing the test/images/** changes from this PR and doing those in another PR with the VERSION bumps I've asked for. The rest of this LGTM, though I agree its size makes this PR prone to getting stuck in rebases

Thanks for the comment. Removed all changes in test/images/*, and rebased the PR.

ahrtr commented:
Rebased.

ahrtr commented:
> I would suggest removing the test/images/** changes from this PR and doing those in another PR with the VERSION bumps I've asked for. The rest of this LGTM, though I agree its size makes this PR prone to getting stuck in rebases

@spiffxp  Just submitted a separate PR [pull/106415](https://github.com/kubernetes/kubernetes/pull/106415 )

ehashman commented:
/remove-sig instrumentation

ehashman reviewed (commented):
/approve

for 
test/e2e/instrumentation/monitoring/stackdriver_metadata_agent.go
test/instrumentation/main.go

marosset commented:
/remove-sig windows
Ping me or add sig-windows back if there are any concerns specific to Windows.

dims commented:
/retest
/assign

ahrtr commented:
> @ahrtr: PR needs rebase.

Just rebased.

marosset commented:
/remove-sig windows

dims commented:
/assign @MadhavJivrajani 

can you please review this Madhav? same as the last one. should be mostly mechanical changes

MadhavJivrajani reviewed (changes requested):
Assuming changes to `images/` are in a seperate PR, there are a few more instances to cover:

- test/e2e_node/kubeletconfig/kubeletconfig.go
- test/e2e_node/device_manager_test.go
- test/integration/apiserver/openapiv3_test.go
- test/integration/auth/podsecurity_test.go

Please also recheck in any other similar PRs that you may have opened, thanks!

ahrtr commented:
> Assuming changes to `images/` are in a seperate PR, there are a few more instances to cover:
> 
>     * test/e2e_node/kubeletconfig/kubeletconfig.go
> 
>     * test/e2e_node/device_manager_test.go
> 
>     * test/integration/apiserver/openapiv3_test.go
> 
>     * test/integration/auth/podsecurity_test.go
> 
> 
> Please also recheck in any other similar PRs that you may have opened, thanks!

Thanks @MadhavJivrajani , just rebased this PR and resolved your comment. PTAL, thanks.

ahrtr commented:
/retest

MadhavJivrajani commented:
/retest

MadhavJivrajani reviewed (approved):
LGTM
Thanks!
This is good to go @dims :)

dims commented:
/skip
/approve
/lgtm
